pkg/config/setup/config.go

```diff
@@ -1316,6 +1316,11 @@ func InitConfig(config pkgconfigmodel.Config) {
 	// TTL refresh period represents how frequently actively detected languages are refreshed by reporting them again to the language detection handler in the cluster agent
 	config.BindEnvAndSetDefault("language_detection.reporting.refresh_period", "20m")
 
+	// config for products implemented in the client libraries
+	config.BindEnv("asm.enabled", "DD_APPSEC_ENABLED_PROPAGATE")
+	config.BindEnv("iast.enabled", "DD_IAST_ENABLED_PROPAGATE")
+	config.BindEnv("sca.enabled", "DD_SCA_ENABLED_PROPAGATE")
```

Comment:
Suggestion:

Since these configurations are only relevant to the admission controller, I think it would be better to have something like:

`admission_controller.auto_instrumentation.asm.enabled` (same for the rest).

[Here](https://github.com/DataDog/datadog-agent/blob/d6a7e7927b96c58b1ca37d5a45453a1a457e3958/pkg/config/setup/config.go#L1085) is an example.

It all depends on whether these configs represent global features (or at least might do so in the future) or not. What I understood is that the scope of these configs is limited to injecting env vars in admission controller only.

WDYT?

Reply:
changed in 4406e81130